Mathlib/Algebra/GroupWithZero/Int.lean

```diff
@@ -0,0 +1,33 @@
+/-
+Copyright (c) 2025 María Inés de Frutos-Fernández, Filippo A. E. Nuccio. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández, Filippo A. E. Nuccio
+-/
+import Mathlib.Algebra.Group.Int.TypeTags
+import Mathlib.Algebra.GroupWithZero.WithZero
+
+/-!
+# Lemmas about `ℤₘ₀`.
+-/
+
+local notation "ℤₘ₀" => WithZero (Multiplicative ℤ)
+
+namespace WithZero
+
+open Multiplicative
+
+theorem ofAdd_zpow (a : ℤ) : (↑(ofAdd a) : ℤₘ₀) = ofAdd (1 : ℤ) ^ a := by
+  rw [← WithZero.coe_zpow, WithZero.coe_inj, ← Int.ofAdd_mul, one_mul]
+
+theorem ofAdd_zpow_zpow_comm (a b c : ℤ) :
+    ((↑(ofAdd a) : ℤₘ₀) ^ b) ^ c = (ofAdd (a : ℤ) ^ c) ^ b := by
+  simp only [← WithZero.coe_zpow]
+  rw [← zpow_mul, mul_comm, zpow_mul]
```

Comment:
```suggestion
  rw [← zpow_mul, ← zpow_mul']
```
This should work for any `DivisionMonoid`

Reply:
The lemma should be 
```
lemma foo {α : Type*} [DivisionMonoid α] (a : α) (m n : ℤ) : 
    (a ^ m) ^ n = (a ^ n) ^ m := by
  rw [← zpow_mul, ← zpow_mul']
```
and it doesn't even need to mention `WithZero` I believe.

Reply:
I moved it to `Algebra.Group.Basic`.